src/Hl7.Fhir.Base/FhirPath/Expressions/SubToken.cs

```diff
@@ -0,0 +1,139 @@
+﻿/* 
+ * Copyright (c) 2015, Firely (info@fire.ly) and contributors
+ * See the file CONTRIBUTORS for details.
+ * 
+ * This file is licensed under the BSD 3-Clause license
+ * available at https://raw.githubusercontent.com/FirelyTeam/firely-net-sdk/master/LICENSE
+ */
+
+using Hl7.Fhir.Language.Debugging;
+using Hl7.Fhir.Utility;
+using Hl7.FhirPath.Sprache;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+
+namespace Hl7.FhirPath.Expressions
+{
+    /// <summary>
+    /// SubTokens are used to represent fragments within a fhirpath expression
+    /// that are a sub-part of the grammar, and are not otherwise present in the
+    /// expression tree as an Expression of some sort.<br/>
+    /// Such as: <code>, . () {} [] '' `` as is</code>
+    /// These do not include functions or binary expressions, which are a part of the execution tree<br/>
+    /// E.g. `as` is a fragment of the typeExpression
+    /// </summary>
+    public class SubToken : IPositionAware<SubToken>
+    {
+        public SubToken(string value, ISourcePositionInfo location = null)
+        {
+            Value = value;
+            Location = location;
+        }
+
+        public SubToken(char value, ISourcePositionInfo location = null)
+        {
+            Value = $"{value}";
+            Location = location;
+        }
+
+        /// <summary>
+        /// The parsed value of this SubToken
+        /// </summary>
+        /// <remarks>
+        /// Derived classes may provide additional information about the value,
+        /// such as the type of whitespace (e.g. comments)
+        /// </remarks>
+        public string Value { get; private set; }
+
+        /// <summary>
+        /// Position information of this token within the complete expression that was parsed
+        /// </summary>
+        public ISourcePositionInfo Location { get; private set; }
+
+        /// <summary>
+        /// Any leading whitespace or comments encountered immediately before this SubToken
+        /// </summary>
+        public IEnumerable<WhitespaceSubToken> leadingWhitespace { get; internal set; }
+
+        /// <summary>
+        /// Any trailing whitespace or comments encountered immediately after this SubToken
+        /// </summary>
+        public IEnumerable<WhitespaceSubToken> trailingWhitespace { get; internal set; }
```

Comment:
Should this be Pascal Cased?

Reply:
Resolved in the other branch when that is merged in too.